integration-test/background/click-to-load-youtube.js

```diff
@@ -0,0 +1,335 @@
+                    'iframe[src*=\'://www.youtube-nocookie.com/embed\']'
+                ],
+                replaceSettings: {
+                    type: 'youtube-video',
+                    buttonText: 'Unblock Video',
+                    infoTitle: 'DuckDuckGo blocked this YouTube Video',
+                    infoText: 'We blocked YouTube from tracking you when the page loaded. If you unblock this Video, YouTube will know your activity.',
+                    simpleInfoText: 'We blocked YouTube from tracking you when the page loaded. If you unblock this Video, YouTube will know your activity.'
+                },
+                clickAction: {
+                    type: 'youtube-video'
+                }
+            },
+            'YouTube embedded subscription button': {
+                selectors: [
+                    'iframe[src*=\'://youtube.com/subscribe_embed\']',
+                    'iframe[src*=\'://youtube-nocookie.com/subscribe_embed\']',
+                    'iframe[src*=\'://www.youtube.com/subscribe_embed\']',
+                    'iframe[src*=\'://www.youtube-nocookie.com/subscribe_embed\']'
+                ],
+                replaceSettings: {
+                    type: 'blank'
+                }
+            }
+        },
+        informationalModal: {},
+        surrogates: [
+            {
+                rule: '(www.)?youtube(-nocookie)?.com/iframe_api',
+                surrogate: 'youtube-iframe-api.js'
+            }
+        ]
+    }
+}
+
+const expectedAPISchema = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../data/api_schemas/youtube-iframe-api.json')).toString())
+const youTubeStandardDomains = new Set(['youtu.be', 'youtube.com', 'www.youtube.com'])
+const youTubeNocookieDomains = new Set(['youtube-nocookie.com', 'www.youtube-nocookie.com'])
+
+let browser
+let bgPage
+let teardown
+
+function summariseYouTubeRequests (requests) {
+    const youTubeIframeApi = { checked: false, alwaysRedirected: true }
+    const youTubeStandard = { blocked: 0, allowed: 0, total: 0 }
+    const youTubeNocookie = { blocked: 0, allowed: 0, total: 0 }
+
+    for (const request of requests) {
+        if (request.url.href === 'https://www.youtube.com/iframe_api') {
+            youTubeIframeApi.alwaysRedirected = (
+                youTubeIframeApi.alwaysRedirected &&
+                request.status === 'redirected' &&
+                request.redirectUrl &&
+                request.redirectUrl.protocol === 'chrome-extension:' &&
+                request.redirectUrl.pathname.endsWith('/youtube-iframe-api.js')
+            )
+            youTubeIframeApi.checked = true
+            continue
+        }
+
+        let stats
+
+        if (youTubeStandardDomains.has(request.url.hostname)) {
+            stats = youTubeStandard
+        } else if (youTubeNocookieDomains.has(request.url.hostname)) {
+            stats = youTubeNocookie
+        } else {
+            continue
```

Comment:
we know that only other expected domain is glitch.me, should we (both here and in FB test) make some noise if we see something unexpected during the test? This could help us catch widget changes